tests/ludwig/modules/test_serialization.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+
+import numpy as np
+import pandas as pd
+import torch
+
+from ludwig.api import LudwigModel
+from ludwig.modules import serialization
+from ludwig.modules.ludwig_module import LudwigModuleState
+from tests.integration_tests.utils import category_feature, generate_data, sequence_feature, text_feature
+
+
+def assert_module_states_equal(a: LudwigModuleState, b: LudwigModuleState):
+    """Recursively asserts that module states a and b are identical."""
+    assert a.type == b.type
+    assert a.ludwig_version == b.ludwig_version
+    assert a.config == b.config
+    assert a.metadata == b.metadata
+    assert sorted(a.saved_weights.keys()) == sorted(b.saved_weights.keys())
+    for ak, av in a.saved_weights.items():
+        assert ak in b.saved_weights
+        bv = b.saved_weights[ak]
+        assert np.allclose(av, bv)
+    assert sorted(a.children.keys()) == sorted(b.children.keys())
+    for ak, ac in a.children.items():
+        assert ak in b.children
+        bc = b.children[ak]
+        assert_module_states_equal(ac, bc)
+
+
+def test_serialize_deserialize_encoder(tmpdir):
+    input_features = [sequence_feature(reduce_output="sum")]
+    output_features = [category_feature(vocab_size=5, reduce_input="sum")]
+    data_csv = generate_data(input_features, output_features, os.path.join(tmpdir, "dataset.csv"))
+    config = {
+        "input_features": input_features,
+        "output_features": output_features,
+        "combiner": {"type": "concat", "output_size": 14},
+    }
+    model_1 = LudwigModel(config)
+    model_1.train(dataset=data_csv, output_directory=tmpdir)
+    # Gets pre-trained encoder.
```

Comment:
nits: Here and in other tests, consider adding whitespace to break up the tests by [Arrange-Act-Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/)